labs/src/main/scala/org/scalalabs/basic/lab01/OOExercise.scala

```diff
@@ -40,6 +40,33 @@ import scala.language.implicitConversions
  *   of type [[org.scalalabs.basic.lab01.CurrencyConverter]]
  * - Use the implicit CurrencyConverter to do the conversion. 
  */
-class Euro {
+abstract class Currency(val symbol: String) {
+}
 
+class Euro(val euro: Int, val cents: Int = 0) extends Currency("EUR") {
+  val inCents: Int = 100 * euro + cents
+  
+  def +(that: Euro) = 
+    Euro.fromCents(inCents + that.inCents)
+  def *(that: Int) =
+    Euro.fromCents(inCents * that)
+   
+  override def toString:String = {
+    def centsToString = if(cents == 0) "--" 
```

Comment:
I like this use of if.

I thought about doing this & inlining it (ultimately did not) - what are your thoughts on inlining it?